crates/ty_server/src/server/api/notifications/did_open.rs

```diff
@@ -53,6 +53,7 @@ impl SyncNotificationHandler for DidOpenTextDocumentHandler {
             }
         }
 
+        publish_settings_diagnostics(session, client);
```

Comment:
Can you tell me why we need to publish settings diagnostics during `did_open` and `did_change`? Is it that the diagnostics show up after starting the server? If so, here might be a better place for it: 

https://github.com/astral-sh/ruff/blob/b4c42eb83b1215ad980a11265f38d84bdeba7e28/crates/ty_server/src/session.rs#L273-L285